### ISSUE TITLE
hgtv unsupported url

### ISSUE THREAD

**Issue by monty124** [ytdl-org/youtube-dl#23092]
## Checklist

- [ x] I'm reporting a broken site support issue
- [ x] I've verified that I'm running youtube-dl version **2019.11.05**
- [ x] I've checked that all provided URLs are alive and playable in a browser
- [x ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [ x] I've searched the bugtracker for similar bug reports including closed ones
- [ x] I've read bugs section in FAQ

## Verbose log

```
 youtube-dl "https://www.hgtv.ca/shows/island-of-bryan/videos/almost-paradise/HGTV0055591700000100/" -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'https://www.hgtv.ca/shows/island-of-bryan/videos/almost-paradise/HGTV0055591700000100/', u'-v']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.11.05
[debug] Python version 2.7.15+ (CPython) - Linux-5.0.0-32-generic-x86_64-with-Ubuntu-18.04-bionic
[debug] exe versions: ffmpeg 3.4.6, ffprobe 3.4.6
[debug] Proxy map: {}
[generic] HGTV0055591700000100: Requesting header
WARNING: Falling back on generic information extractor.
[generic] HGTV0055591700000100: Downloading webpage
[generic] HGTV0055591700000100: Extracting information
ERROR: Unsupported URL: https://www.hgtv.ca/shows/island-of-bryan/videos/almost-paradise/HGTV0055591700000100/
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/generic.py", line 2373, in _real_extract
    doc = compat_etree_fromstring(webpage.encode('utf-8'))
  File "/usr/local/bin/youtube-dl/youtube_dl/compat.py", line 2551, in compat_etree_fromstring
    doc = _XML(text, parser=etree.XMLParser(target=_TreeBuilder(element_factory=_element_factory)))
  File "/usr/local/bin/youtube-dl/youtube_dl/compat.py", line 2540, in _XML
    parser.feed(text)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1659, in feed
    self._raiseerror(v)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1523, in _raiseerror
    raise err
ParseError: not well-formed (invalid token): line 96, column 71
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/generic.py", line 3354, in _real_extract
    raise UnsupportedError(url)
UnsupportedError: Unsupported URL: https://www.hgtv.ca/shows/island-of-bryan/videos/almost-paradise/HGTV0055591700000100/

```

## Description

as of late it seems hgtv is no longer supported, I'm unable to tell if they have changed something within the rendered html but tests to different videos all seem to generate the same error.
hgtv.ca is region locked.

**Comment by dwalkerdon** [comment]
A&E.com is no longer supported as well, i tried to download this https://play.aetv.com/shows/beyond-scared-straight/season-1/episode-1 video from the site, but youtube-dl doesn't detect it

**Comment by tmgd-dot** [comment]
I managed to get one episode to download from aetv by messing around with Chrome dev tools and plugging the URL into ytdl, but can't for the life of me figure out where it came from - might help someone

https://aetv.global.ssl.fastly.net/token/xxxxxxxU=/AETN-AETV_VMS/BRAND_AEN_SPPW_77582_CRS_2398_030_20140421_V1_S3_1800.mp4.m3u8

**Comment by tv21** [comment]
In the original issue I note the line that says:

[debug] Python version 2.7.15+ (CPython) - Linux-5.0.0-32-generic-x86_64-with-Ubuntu-18.04-bionic

The version of Python may be the problem.  One thing to try with that and similar sites (all the sites listed under "WATCH MORE FULL EPISODES FROM THESE NETWORKS" at the bottom of the page) is to force the use of python3, assuming you have it on your system (enter "which python3" at a command prompt and it should show a path if you have it installed).  If you do, then try this, go to the site in a browser as if you were going to play it in a web browser, log in with your provider, and use a cookies.txt extension (such as https://chrome.google.com/webstore/detail/cookiestxt/njabckikapfpffapmjgojcnbfjonfjfg for Chrome) to generate a cookies.txt file.  Then try this:

python3 /usr/local/bin/youtube-dl --cookies cookies.txt https://...

Note that you must use the full path to youtube-dl and you may need to include the path to the cookies.txt file as well.  Thanks to remitamine for mentioning the Python version problem in another issue.

**Comment by remitamine** [comment]
this issue(support for new Canadian HGTV URLs) has been fixed in the latest version(2019.11.28).
the AETV issue is completely unrelated, you can open a new issue for it.
